homeassistant/components/tesla_wall_connector/__init__.py

```diff
@@ -0,0 +1,185 @@
+        except WallConnectorConnectionError as ex:
+            raise UpdateFailed(
+                f"Could not fetch data from Tesla WallConnector at {hostname}: Cannot connect"
+            ) from ex
+        except WallConnectorError as ex:
+            raise UpdateFailed(
+                f"Could not fetch data from Tesla WallConnector at {hostname}: {ex}"
+            ) from ex
+
+        if previous_contactor_closed != vitals.contactor_closed:
+            coordinator.update_interval = (
+                timedelta(seconds=poll_interval)
+                if not vitals.contactor_closed
+                else timedelta(seconds=poll_interval_charging)
+            )
+            _LOGGER.debug(
+                "Contactor closed: %s. Update interval: %s",
+                str(vitals.contactor_closed),
+                str(coordinator.update_interval),
+            )
+
+        return {
+            WALLCONNECTOR_DATA_VITALS: vitals,
+            WALLCONNECTOR_DATA_LIFETIME: lifetime,
+        }
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="tesla-wallconnector",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=poll_interval),
+    )
+
+    hass.data[DOMAIN][entry.entry_id][
+        WALLCONNECTOR_DATA_UPDATE_COORDINATOR
+    ] = coordinator
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+def prefix_entity_name(name: str) -> str:
+    """Prefixes entity name."""
+    return f"{WALLCONNECTOR_DEVICE_NAME} {name}"
+
+
+def get_unique_id(serial_number: str, key: str) -> str:
+    """Get a unique entity name."""
+    return f"{serial_number}-{key}"
+
+
+class WallConnectorEntity(CoordinatorEntity):
+    """Base class for Wall Connector entities."""
+
+    def __init__(self, wall_connector: dict) -> None:
+        """Initialize WallConnector Entity."""
+        self.wall_connector = wall_connector
+        self._attr_unique_id = get_unique_id(
+            wall_connector[WALLCONNECTOR_SERIAL_NUMBER], self.entity_description.key
+        )
+        super().__init__(wall_connector[WALLCONNECTOR_DATA_UPDATE_COORDINATOR])
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return information about the device."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.wall_connector[WALLCONNECTOR_SERIAL_NUMBER])},
+            default_name=WALLCONNECTOR_DEVICE_NAME,
+            model=self.wall_connector[WALLCONNECTOR_PART_NUMBER],
+            sw_version=self.wall_connector[WALLCONNECTOR_FIRMWARE_VERSION],
+            default_manufacturer="Tesla",
```

Comment:
Is there a web ui? If so add `configuration_url`

Reply:
Yes there is, but it's just not useful for anything as configuration is done via built in wifi hotspot:
![image](https://user-images.githubusercontent.com/1431687/142764876-ebf50488-ca9b-4d0c-95ac-79f04127d5db.png)